src/trio/_core/_tests/test_run.py

```diff
@@ -178,16 +181,15 @@ async def main() -> None:
             nursery.start_soon(looper)
             nursery.start_soon(crasher)
 
-    with pytest.raises(ValueError) as excinfo:
+    with pytest.raises(ExpectedExceptionGroup(ValueError("argh"))):
```

Comment:
Hm. Earlier you checked `ExpectedExceptionGroup(ValueError)` then checked `excinfo.value.exceptions[0] is exc` but wouldn't that be `ExpectedExceptionGroup(exc)` to be similar to this?

Probably not because `is` vs `==`? (I am being very confused right now :( )

Reply:
yeah, `ExpectedExceptionGroup` checks with `isinstance` - but this test previously did an identity check w/ the caught exception so I thought I'd keep that check. Will add a comment

Most tests can't check with identity as they're not the one raising, so makes some sense that this one does when it can, to check that a different exception with the same type didn't get raised

Reply:
in this case this is more of a question of the behavior of ExpectedExceptionGroup/RaisesGroup. But once the RaisesGroup PR is merged and I rebase this on top of it I will rewrite the former to do
```python
with RaisesGroup(Matcher(ValueError, check=lambda x: x is exc)):
  ...
```